src/util/transactionUtils.ts

```diff
@@ -71,3 +77,211 @@ export function getTransactionDisplayPayload({
   transactionDisplayPayload.spec_version = spec_version;
   return { transactionDisplayPayload, payloadHex };
 }
+
```

Comment:
You can add the following array and function that will simplify a lot the `updatedSteps` and `createEmptySteps` (see below where updateSteps and startSteps are) :

```suggestion

const steps = [
  ['include-message-block', 'Include message in block'],
  ['finalized-block', 'Finalize block'],
  ['relay-block', 'Relay block'],
  ['deliver-message-block', 'Deliver message in target block'],
  ['finalized-message', 'Finalize message'],
  ['confirm-delivery', 'Confirm delivery']
];

const step = (step: number, chainType: string, status?: TransactionStatusEnum, labelOnChain?: any) => {
  const obj = {
    id: 'test-step-' + steps[step - 1][0],
    chainType,
    label: steps[step - 1][1],
    status: status || TransactionStatusEnum.NOT_STARTED
  };
  if (!isEmpty(labelOnChain)) {
    return { ...obj, labelOnChain };
  }
  return obj;
};

```

Reply:
wow classy! let me play with it ... i think this will go in thanks!